components/Shipping/basic_usage.rst

```diff
@@ -0,0 +1,657 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+**DelegatingCalculator** class delegates the calculation of charge for particular shipping subject to a correct calculator instance,
+based on the name defined on the shipping method. It uses **CalculatorRegistry** to keep all calculators registered inside
+container. The calculators are retrieved by name.
+
+.. code-block:: php
+
+    <?php
+
+        use Sylius\Component\Shipping\Model\ShippingMethod;
+        use Sylius\Component\Shipping\Calculator\DefaultCalculators;
+        use Sylius\Component\Shipping\Calculator\PerItemRateCalculator;
+        use Sylius\Component\Shipping\Calculator\FlexibleRateCalculator;
+        use Sylius\Component\Shipping\Model\Shipment;
+        use Sylius\Component\Shipping\Model\ShipmentItem;
+        use Sylius\Component\Shipping\Calculator\Registry\CalculatorRegistry;
+        use Sylius\Component\Shipping\Calculator\DelegatingCalculator;
+
+        $configuration = array(
+            'first_item_cost'       => 1000,
+            'additional_item_cost'  => 200,
+            'additional_item_limit' => 2
+        );
+        $shippingMethod = new ShippingMethod();
+        $shippingMethod->setConfiguration($configuration);
+        $shippingMethod->setCalculator(DefaultCalculators::FLEXIBLE_RATE);
+
+        $shipmentItem = new ShipmentItem();
+
+        $shipment = new Shipment();
+        $shipment->setMethod($shippingMethod);
+        $shipment->addItem($shipmentItem);
+
+        $calculatorRegistry = new CalculatorRegistry();
+        $flexibleRateCalculator = new FlexibleRateCalculator();
+        $perItemRateCalculator = new PerItemRateCalculator();
+        $calculatorRegistry
+        ->registerCalculator(DefaultCalculators::FLEXIBLE_RATE, $flexibleRateCalculator);
+        $calculatorRegistry
+        ->registerCalculator(DefaultCalculators::PER_ITEM_RATE, $perItemRateCalculator);
+
+        $delegatingCalculators = new DelegatingCalculator($calculatorRegistry);
+        $delegatingCalculators->calculate($shipment); // returns 1000
+
+        $configuration2 = array('amount' => 200);
+        $shippingMethod2 = new ShippingMethod();
+        $shippingMethod2->setConfiguration($configuration2);
+        $shippingMethod2->setCalculator(DefaultCalculators::PER_ITEM_RATE);
+
+        $shipment->setMethod($shippingMethod2);
+        $delegatingCalculators->calculate($shipment); // returns 200
+
+.. caution::
+       The method ``->registerCalculator()`` throws `ExistingCalculatorException`_.
+       The method ``->getCalculator()`` used in ``->calculate`` throws `NonExistingCalculatorException`_.
+
+.. _ExistingCalculatorException: http://api.sylius.org/Sylius/Component/Shipping/Calculator/Registry/ExistingCalculatorException.html
+.. _NonExistingCalculatorException: http://api.sylius.org/Sylius/Component/Shipping/Calculator/Registry/NonExistingCalculatorException.html
+
+FlatRateCalculator
+~~~~~~~~~~~~~~~~~~
```

Comment:
We should have all calculators in a separate chapter (file) of these docs? What do you think?